### PR TITLE
Model reorganization effort attempt 1

### PR DESCRIPTION
**Pull request by chris-grove**
- used allele.yaml as a pilot case
- Moved all DTO classes and DTO-specific slots to allele_dto.yaml
- Alphabetized all class and slot definitions within their respective sections

### CODE REVIEW

File: model/schema/allele_dto.yaml [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+  transgene_chromosome_location_curie:
+    description: >-
+      The curie string of the chromosome to which a transgene has been mapped. Used for WormBase
+      transgenes that have been integrated into the genome and mapped to a chromosome.
+    domain: AlleleDTO
+    range: string
+    multivalued: false
+
+## ------------
+## ALLELE DTO SLOT ANNOTATION SLOTS
```
Comment (markquintontulloch) [inline model/schema/allele_dto.yaml:450]: I’m not sure we should separate these from other slots.  It’s not immediately obvious from the slot name that these slots have a range of a SlotAnnotation.  Also, these are slots of the Allele class not the SlotAnnotation classes - my interpretation of ‘Slot Annotation Slots’ would be slots used in the SlotAnnotation classes themselves (eg. mutation_type_curies).  I think just listing all slots alphabetically would be more intuitive.

Comment (chris-grove) [reply model/schema/allele_dto.yaml]: @markquintontulloch 
You're right, sorry, I thought it made a lot more sense earlier when I was trying to compare the DTO and non-DTO yaml files side-by-side. I've fixed in my latest commit.

Comment (chris-grove) [reply model/schema/allele_dto.yaml]: Actually, not all done yet...

Comment (chris-grove) [reply model/schema/allele_dto.yaml]: OK, now fixed

**Review by markquintontulloch — APPROVED** [review]
Looks good.  As a side note unrelated to this PR, at some point I think we should go through and give the predicate_name slots.more descriptive names to make it clearer what is expected (as we have for disease_relation_name in the DiseaseAnnotationDTO class)

**Comment by chris-grove** [comment]
@markquintontulloch 
Yep! Sounds good. Thanks!

**Pull request merged**